pkg/fakerp/admin_handlers.go

```diff
@@ -106,71 +96,93 @@ func (s *Server) handleListBackups(w http.ResponseWriter, req *http.Request) {
 
 // handleRestore handles admin requests to restore an etcd cluster from a backup
 func (s *Server) handleRestore(w http.ResponseWriter, req *http.Request) {
-	cs := s.read()
-	if cs == nil {
+	cs, err := s.store.Get()
+	if cs == nil || err != nil {
 		s.internalError(w, "Failed to read the internal config")
 		return
 	}
 
 	backupName := chi.URLParam(req, "backupName")
 
-	pluginErr := s.plugin.RecoverEtcdCluster(req.Context(), cs, GetDeployer(s.log, cs, s.testConfig), backupName)
-	var err error
-	if pluginErr != nil {
-		// TODO: fix this nastiness: https://golang.org/doc/faq#nil_error
```

Comment:
on what basis are you removing this code?  I don't believe this PR is fixing the underlying issue, so it can't go.